lib/rules/logical-assignment-operators.js

```diff
@@ -0,0 +1,465 @@
+                            // ?? and || / && cannot be mixed but have same precedence
+                            const mixed = assignment.operator === "??=" && astUtils.isLogicalExpression(assignment.right);
+
+                            if (!astUtils.isParenthesised(sourceCode, assignment.right) && (precedence || mixed)) {
+
+                                // -> foo = foo || (bar)
+                                yield ruleFixer.insertTextBefore(assignment.right, "(");
+                                yield ruleFixer.insertTextAfter(assignment.right, ")");
+                            }
+                        }
+                    };
+
+                    context.report(createConditionalFixer(descriptor, suggestion, cannotBeGetter(assignment.left)));
+                }
+            };
+        }
+
+        return {
+
+            // foo = foo || bar
+            "AssignmentExpression[operator='='][right.type='LogicalExpression']"(assignment) {
+                if (!astUtils.isSameReference(assignment.left, assignment.right.left, true)) {
+                    return;
+                }
+
+                const descriptor = {
+                    messageId: "assignment",
+                    node: assignment,
+                    data: { operator: assignment.right.operator }
+                };
+                const suggestion = {
+                    messageId: "useLogicalOperator",
+                    data: { operator: assignment.operator },
+                    *fix(ruleFixer) {
+                        if (sourceCode.getCommentsInside(assignment).length > 0) {
+                            return;
+                        }
+
+                        // No need for parenthesis around the assignment based on precedence as the precedence stays the same even with changed operator
+                        const assignmentOperatorToken = getOperatorToken(assignment);
+
+                        // -> foo ||= foo || bar
+                        yield ruleFixer.insertTextBefore(assignmentOperatorToken, assignment.right.operator);
+
+                        // -> foo ||= bar
+                        yield ruleFixer.removeRange([assignment.right.range[0], assignment.right.right.range[0]]);
+
+                        if (astUtils.isParenthesised(sourceCode, assignment.right.right)) {
+
+                            // Opening parenthesis already removed
+                            const closingParenthesis = sourceCode.getTokenAfter(assignment.right.right);
+
+                            yield ruleFixer.remove(closingParenthesis);
+                        }
+                    }
+                };
+
+                context.report(createConditionalFixer(descriptor, suggestion, cannotBeGetter(assignment.left)));
+            },
+
+            // foo || (foo = bar)
+            'LogicalExpression[right.type="AssignmentExpression"][right.operator="="]'(logical) {
+                if (astUtils.isSameReference(logical.left, logical.right.left)) {
```

Comment:
```suggestion
                if (isReference(logical.left) && astUtils.isSameReference(logical.left, logical.right.left)) {
```

To avoid false positives such as this one:

```js
/*eslint logical-assignment-operators: ["error", "always"]*/

a?.b && (a.b = c) // false positive as `a.b &&= c` is not equivalent

```